assets/js/redirect.js

```diff
@@ -3,26 +3,16 @@
 		const publishBtn = $('#publish');
 		const fromRule = $('#srm_redirect_rule_from');
 		const toRule = $('#srm_redirect_rule_to');
+		let timer = 0;
+		let currentRequest = null;
 
-		fromRule.change(function(el) {
-			publishBtn.prop('disabled', true);
-			$.get(
-				window.ajaxurl,
-				{
-					action: 'srm_validate_from_url',
-					from: fromRule.val(),
-					_wpnonce: $('#srm_redirect_nonce').val()
-				}
-			).done(function( data ) {
-				if ( '1' === data ) {
-					$('#message').html( '' ).hide();
-					publishBtn.prop('disabled', false);
-				} else if ( '0' === data ) {
-					$('#message').html( `<p>${redirectValidation.urlError}</p>` ).show();
-				} else {
-					$('#message').html( `<p>${redirectValidation.fail.replace( '%s', data )}</p>` ).show();
-				}
-			});
+		fromRule.keyup(function(el) {
```

Comment:
```suggestion
		fromRule.on('input', function(el) {
```

I think the [`input`](https://developer.mozilla.org/en-US/docs/Web/API/Element/input_event) event is better here as it describes the exact use case: 

> The input event fires when the value of an [`<input>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input), [`<select>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select), or [`<textarea>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/textarea) element has been changed as a direct result of a user action (such as typing in a textbox or checking a checkbox).

Reply:
Thank you for this. I've made this change now.